### PR TITLE
AP_BattMonitor: warn if battery backend type not found

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: libraries/AP_BattMonitor/AP_BattMonitor.cpp [hunk 1]
```diff
@@ -147,7 +147,9 @@ AP_BattMonitor::init()
                 drivers[instance] = new AP_BattMonitor_FuelLevel_PWM(*this, state[instance], _params[instance]);
                 break;
             case AP_BattMonitor_Params::BattMonitor_TYPE_NONE:
+                break;
             default:
+                gcs().send_text(MAV_SEVERITY_WARNING, "Unknown battery monitor type %u", (uint8_t)get_type(instance));
```
Comment (WickedShell) [inline libraries/AP_BattMonitor/AP_BattMonitor.cpp]: Ehhhhh? Not sure this is actually worth it. This occurs early enough in the boot process most people won't see it, and you can actually diagnose this by the `BATTERY_STATUS` messages.

Comment (peterbarker) [reply libraries/AP_BattMonitor/AP_BattMonitor.cpp]: In actual fact it doesn't come out at all ATM as we haven't initialised the GCS_MAVLink objects at this point....

But at least this way we can say we tried.

And I do plan on fixing the "not emitted" thing.

Looking at `BATTERY_STATUS` is probably a reasonable comeback, 'though.

Comment (WickedShell) [reply libraries/AP_BattMonitor/AP_BattMonitor.cpp]: Yeah if it's not gonna come out, lets not waste the flash on it :)

With that being said doesn't it end up in the queue of statustexts in the GCS, it just doesn't get serviced/may be pushed out by others?

Comment (peterbarker) [reply libraries/AP_BattMonitor/AP_BattMonitor.cpp]: Nah.  At this point no channel is active or streaming, so the mask is empty.

**Pull request closed without merging**